machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,340 @@
+
+	// Tags are the set of metadata to add to an instance.
+	// +optional
+	Tags []Tag `json:"tag,omitempty"`
+}
+
+// ResourceTagReference is a reference to a specific AlibabaCloud resource by ID, or tags.
+// Only one of ID or Tags may be specified. Specifying more than one will result in
+// a validation error.
+type AlibabaResourceReference struct {
+	// ID of resource
+	// +optional
+	ID string `json:"id,omitempty"`
+
+	// Tags is a set of metadata based upon ECS object tags used to identify a resource
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlibabaCloudMachineProviderConfigList contains a list of AlibabaCloudMachineProviderConfig
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AlibabaCloudMachineProviderConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AlibabaCloudMachineProviderConfig `json:"items"`
+}
+
+// AlibabaCloudMachineProviderStatus is the Schema for the alibabacloudmachineproviderconfig API
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type AlibabaCloudMachineProviderStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// InstanceID is the instance ID of the machine created in alibabacloud
+	// +optional
+	InstanceID *string `json:"instanceId,omitempty"`
+
+	// InstanceState is the state of the alibabacloud instance for this machine
+	// +optional
+	InstanceState *string `json:"instanceState,omitempty"`
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// SystemDiskProperties contains the information regarding the system disk including performance, size, name, and category
+type SystemDiskProperties struct {
+	// Category is the category of the system disk.
+	// Valid values:
+	// cloud_essd: ESSD. When the parameter is set to this value, you can use the SystemDisk.PerformanceLevel parameter to specify the performance level of the disk.
+	// cloud_efficiency: ultra disk.
+	// cloud_ssd: standard SSD.
+	// cloud: basic disk.
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently for non-I/O optimized instances of retired instance types, the default is `cloud`.
+	// Currently for other instances, the default is `cloud_efficiency`.
+	// +kubebuilder:validation:Enum="AlibabaDiskCatagoryUltraDisk"; "AlibabaDiskCategorySSD"; "AlibabaDiskCategoryESSD"; "AlibabaDiskCategoryBasic"
```

Comment:
these enum values should be the actual value from the const, not the name of the const variable